src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
+const allAnimal = [];
+
 class Animal {
-  // write your code here
+  static alive() {
+    return allAnimal.filter(item => item.health > 0);
+  }
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+    allAnimal.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    allAnimal.push(this);
+  }
+
+  bite(name) {
+    if (name.hidden === true || name instanceof Carnivore) {
```

Comment:
```suggestion
    if (name.hidden || name instanceof Carnivore) {
```
 redundant to compare with boolean